src/lib/utils/SynapseContext.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext } from 'react'
+import { QueryClient, QueryClientProvider } from 'react-query'
+
+const defaultQueryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      staleTime: 30 * 1000, // 30s
+      retry: false, // SynapseClient knows which queries to retry
+    },
+  },
+})
+
+export type SynapseContextType = {
+  accessToken?: string
+  isInExperimentalMode: boolean
+  utcTime: boolean
+}
+
+export const SynapseContext = React.createContext<
+  SynapseContextType | undefined
+>(undefined)
+
+export type SynapseContextProviderProps = {
+  synapseContext?: SynapseContextType
+  queryClient?: QueryClient
+}
+
+/**
+ * Provides context necessary for most components in SRC
+ * @param param0
+ * @returns
+ */
+export const SynapseContextProvider: React.FunctionComponent<SynapseContextProviderProps> = ({
```

Comment:
Optional reading: [here's a blog post](https://kentcdodds.com/blog/how-to-use-react-context-effectively) by Kent C. Dodds that has guided many of the design decisions made here.